web_src/js/features/notification.js

```diff
@@ -4,6 +4,30 @@ const {appSubUrl, csrfToken, notificationSettings, assetVersionEncoded} = window
 let notificationSequenceNumber = 0;
 
 export function initNotificationsTable() {
+  const table = document.getElementById('notification_table');
+  if (!table) return;
+
+  // when page restores from bfcache, delete previously clicked items
+  window.addEventListener('pageshow', (e) => {
+    if (e.persisted) { // page was restored from bfcache
+      const table = document.getElementById('notification_table');
+      const unreadCountEl = document.querySelector('.notifications-unread-count');
+      let unreadCount = parseInt(unreadCountEl.textContent);
+      for (const item of table.querySelectorAll('.notifications-item[data-remove="true"]')) {
+        item.remove();
+        unreadCount -= 1;
+      }
+      unreadCountEl.textContent = unreadCount;
+    }
+  });
+
+  // mark clicked unread links for deletion on bfcache restore
+  for (const link of table.querySelectorAll('.notifications-item[data-status="1"] .notifications-link')) {
+    link.addEventListener('click', (e) => {
+      e.target.closest('.notifications-item').setAttribute('data-remove', 'true');
+    }, {capture: true});
```

Comment:
Not sure why it needs `capture: true` , I guess normal bubbling still works?

Reply:
Normal works, but I thought some browsers might trigger the navigation to the link before a bubble phase listener, in which case it would not run. Not sure about the actual specced behaviour on this, likely the navigation happens after the bubble listener phase.

Reply:
IMO that's impossible. It can always do `e.preventDefault` in a normal click event listener, without "capture".

If you suspect that "some browsers might trigger the navigation", there will be already a lot of bugs.

Reply:
Removed